Framework/MSBuild.exe publish fails when using default local registry

```sh-session
❯ msbuild -restore -bl -t:PublishContainer -p:PublishProfile=DefaultContainer
PublishContainer:
  C:\Program Files\dotnet\dotnet.exe "C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.3.0-alpha.32.gcb428eb1b2\build\..\container 
  ize\containerize.dll" "bin\Debug\net7.0\linux-x64\publish" --baseregistry mcr.microsoft.com --baseimagename dotnet/runtime-deps --baseimagetag 7 
  .0 --outputregistry  --imagename container-package-test --workingdirectory /app --entrypoint /app/container-package-test --labels org.opencontai 
  ners.image.created="2023-01-19T16:15:51.9470176Z" --imagetags "1.0.0"
  Option '--imagename' is required.
  Unrecognized command or argument 'container-package-test'.

C:\Users\raines\.nuget\packages\microsoft.net.build.containers\0.3.0-alpha.32.gcb428eb1b2\build\Microsoft.NET.Build.Containers.targets(116,9): error MSB6006: "dotnet.exe" exited with code 1. [S:\play\container-package-test\container-package-test.csproj]
```

(when no real registry is set via property).

--- a/Microsoft.NET.Build.Containers/ContainerBuilder.cs
+++ b/Microsoft.NET.Build.Containers/ContainerBuilder.cs
@@ -7,7 +7,7 @@ using System.Threading.Tasks;
 
 public static class ContainerBuilder
 {
-    public static async Task Containerize(DirectoryInfo folder, string workingDir, string registryName, string baseName, string baseTag, string[] entrypoint, string[] entrypointArgs, string imageName, string[] imageTags, string outputRegistry, string[] labels, Port[] exposedPorts, string[] envVars)
+    public static async Task Containerize(DirectoryInfo folder, string workingDir, string registryName, string baseName, string baseTag, string[] entrypoint, string[] entrypointArgs, string imageName, string[] imageTags, string? outputRegistry, string[] labels, Port[] exposedPorts, string[] envVars)
     {
         var isDockerPull = String.IsNullOrEmpty(registryName);
         if (isDockerPull) {
@@ -30,7 +30,7 @@ public static class ContainerBuilder
         img.SetEntrypoint(entrypoint, entrypointArgs);
 
         var isDockerPush = String.IsNullOrEmpty(outputRegistry);
-        Registry? outputReg = isDockerPush ? null : new Registry(ContainerHelpers.TryExpandRegistryToUri(outputRegistry));
+        Registry? outputReg = isDockerPush ? null : new Registry(ContainerHelpers.TryExpandRegistryToUri(outputRegistry!));
 
         foreach (var label in labels)
         {

--- a/Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs
+++ b/Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs
@@ -75,7 +75,7 @@ public partial class CreateNewImage : ToolTask
                " --baseregistry " + BaseRegistry +
                " --baseimagename " + BaseImageName +
                " --baseimagetag " + BaseImageTag +
-               " --outputregistry " + OutputRegistry +
+               (OutputRegistry is not null ? " --outputregistry " + OutputRegistry : "") +
                " --imagename " + ImageName +
                " --workingdirectory " + WorkingDirectory +
                (Entrypoint.Length > 0 ? " --entrypoint " + String.Join(" ", Entrypoint.Select((i) => i.ItemSpec)) : "") +

--- a/containerize/Program.cs
+++ b/containerize/Program.cs
@@ -32,7 +32,7 @@ var outputRegistryOpt = new Option<string>(
     name: "--outputregistry",
     description: "The registry to push to.")
 {
-    IsRequired = true
+    IsRequired = false
 };
 
 var imageNameOpt = new Option<string>(
@@ -175,7 +175,7 @@ root.SetHandler(async (context) =>
     string _baseReg = context.ParseResult.GetValueForOption(baseRegistryOpt) ?? "";
     string _baseName = context.ParseResult.GetValueForOption(baseImageNameOpt) ?? "";
     string _baseTag = context.ParseResult.GetValueForOption(baseImageTagOpt) ?? "";
-    string _outputReg = context.ParseResult.GetValueForOption(outputRegistryOpt) ?? "";
+    string? _outputReg = context.ParseResult.GetValueForOption(outputRegistryOpt);
     string _name = context.ParseResult.GetValueForOption(imageNameOpt) ?? "";
     string[] _tags = context.ParseResult.GetValueForOption(imageTagsOpt) ?? Array.Empty<string>();
     string _workingDir = context.ParseResult.GetValueForOption(workingDirectoryOpt) ?? "";

